File: kotlin-node/karakum/src/jsMain/resources/plugins/convertAssertStrict.js
import ts from "typescript"
import * as karakum from "../karakum.mjs"

export default function (node, context, render) {
    const sourceFileName = node.getSourceFile()?.fileName ?? "generated.d.ts"

    if (
        sourceFileName.endsWith("assert.d.ts")
        && ts.isVariableDeclaration(node)
        && ts.isIdentifier(node.name)
        && node.name.text === "strict"
    ) {
        return ""
    }

    return null
}

File: kotlin-node/karakum/src/jsMain/kotlin/node/karakum/plugins/convertAssertStrict.kt
package node.karakum.plugins

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.createSimplePlugin
import io.github.sgrishchenko.karakum.util.getSourceFileOrNull
import typescript.isIdentifier
import typescript.isVariableDeclaration

val convertAssertStrict = createSimplePlugin { node, _, _ ->
    nullable {
        val sourceFileName = ensureNotNull(node.getSourceFileOrNull()?.fileName)

        ensure(sourceFileName.endsWith("assert.d.ts"))
        ensure(isVariableDeclaration(node))

        val name = node.name
        ensure(isIdentifier(name))

        ensure(name.text === "strict")

        ""
    }
}
